tf_trees/neural_trees_helpers.cc

```diff
@@ -67,7 +55,8 @@ void ForwardPassSingleSample(const Eigen::MatrixXf& node_weights,
                              const int depth, const float smooth_step_param,
                              Eigen::VectorXf* output,
                              std::vector<Node>* tree_nodes,
-                             std::vector<int>* reachable_leaves) {
+                             std::vector<int>* reachable_leaves,
+                             int* num_reachable_leaves, bool training_mode) {
```

Comment:
move the training mode before the pointers * so all const things are first
const bool training+mode
Why do you need num_reachable_leaves if you have vector of reachable leaves (len of it is the number no?)